src/Files.App/Utils/Git/GitHelpers.cs

```diff
@@ -108,30 +121,54 @@ public static string GetOriginRepositoryName(string? path)
 			return repositoryName[..repositoryName.LastIndexOf(".git")];
 		}
 
-		public static BranchItem[] GetBranchesNames(string? path)
+		public static async Task<BranchItem[]> GetBranchesNames(string? path)
 		{
 			if (string.IsNullOrWhiteSpace(path) || !Repository.IsValid(path))
 				return Array.Empty<BranchItem>();
 
-			using var repository = new Repository(path);
+			_owningThread ??= new ThreadWithMessageQueue();
+			
+			lock (_operationsCountLock)
+				++_activeOperationsCount;
```

Comment:
It is better to use `Interlocked` class.
https://learn.microsoft.com/dotnet/api/system.threading.interlocked?view=net-7.0

Reply:
I used `Volatile.Read()` to check the value. Is it ok or should I use `Interlocked.Read()`?

Reply:
See my suggestion.